browser/brave_rewards/rewards_panel_coordinator.cc

```diff
@@ -0,0 +1,102 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/browser/brave_rewards/rewards_panel_coordinator.h"
+
+#include <string>
+#include <utility>
+
+#include "base/feature_list.h"
+#include "brave/browser/brave_rewards/rewards_panel_extension_handler.h"
+#include "brave/browser/profiles/profile_util.h"
+#include "brave/components/brave_rewards/common/features.h"
+#include "brave/components/constants/webui_url_constants.h"
+#include "chrome/browser/profiles/profile.h"
+#include "chrome/browser/ui/browser_window.h"
+
+namespace brave_rewards {
+
+RewardsPanelCoordinator::RewardsPanelCoordinator(Browser* browser)
+    : BrowserUserData<RewardsPanelCoordinator>(*browser) {
+  // If we are using the Rewards extension to display the Rewards panel, then
+  // create an instance of `RewardsPanelExtensionHandler` to load the extension
+  // and dispatch panel requests to the extension.
+  if (!base::FeatureList::IsEnabled(features::kWebUIPanelFeature)) {
+    auto handler =
+        std::make_unique<RewardsPanelExtensionHandler>(&GetBrowser());
+    AddObserver(handler.get());
+    extension_handler_ = std::move(handler);
+  }
+}
+
+RewardsPanelCoordinator::~RewardsPanelCoordinator() = default;
+
+bool RewardsPanelCoordinator::IsRewardsPanelURLForTesting(const GURL& url) {
+  if (base::FeatureList::IsEnabled(features::kWebUIPanelFeature)) {
+    return url.host() == kBraveRewardsPanelHost;
+  } else {
+    return RewardsPanelExtensionHandler::IsRewardsExtensionPanelURL(url);
+  }
+}
+
+RewardsPanelCoordinator* RewardsPanelCoordinator::RegisterForBrowser(
+    Browser* browser) {
+  DCHECK(browser);
+  if (brave::IsRegularProfile(browser->profile())) {
```

Comment:
Duplicating this check is multiple places is fragile and this doesn't actually seem necessary because nothing is going to access this in other windows. Also it's a bit confusing imo to add a separate static method for creating this. I would either just call `RewardsPanelCoordinator::CreateForBrowser` in the brave browser view and not worry about it or create brave_rewards::IsAllowedForContext similar to brave_wallet and wrap it around the call to `RewardsPanelCoordinator::CreateForBrowser`

Reply:
Added a `rewards_context_utils` as suggested and had `BraveBrowserView` use it.